swagger/che-swagger-module/src/main/java/org/eclipse/che/swagger/deploy/BasicSwaggerConfigurationModule.java

```diff
@@ -10,24 +10,48 @@
  */
 package org.eclipse.che.swagger.deploy;
 
+import static java.util.Objects.requireNonNull;
+
 import com.google.common.collect.ImmutableMap;
 import com.google.inject.servlet.ServletModule;
+import java.io.File;
+import java.net.URL;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import org.slf4j.LoggerFactory;
 
 /**
  * Module provide basic swagger configuration
  *
  * @author Sergii Kabashnyuk
  */
 public class BasicSwaggerConfigurationModule extends ServletModule {
+  private static final org.slf4j.Logger LOG =
+      LoggerFactory.getLogger(BasicSwaggerConfigurationModule.class);
+
   @Override
   protected void configureServlets() {
     bind(io.swagger.jaxrs.config.DefaultJaxrsConfig.class).asEagerSingleton();
     serve("/swaggerinit")
         .with(
             io.swagger.jaxrs.config.DefaultJaxrsConfig.class,
             ImmutableMap.of(
-                "api.version", "1.0",
+                "api.version", getCheVersion(),
                 "swagger.api.title", "Eclipse Che",
                 "swagger.api.basepath", "/api"));
   }
+
+  private String getCheVersion() {
+    try {
+      URL url =
+          BasicSwaggerConfigurationModule.class.getProtectionDomain().getCodeSource().getLocation();
+      try (JarFile jar = new JarFile(new File(url.toURI()))) {
+        final Manifest manifest = requireNonNull(jar.getManifest(), "Manifest must not be null");
+        return manifest.getMainAttributes().getValue("Specification-Version");
+      }
+    } catch (Exception e) {
+      LOG.warn("Unable to retrieve implementation version from manifest file", e);
+      return "unknown";
```

Comment:
Up https://github.com/eclipse/che-lib/pull/92#discussion_r202915086 Not to lose =)

Reply:
It may be left as "unknown" since RuntimeException here fails the only swagger.war but not whole tomcat, so there is still a possibility to miss failed was the same as a logged error.

But please, use LOG.error instead of LOG.warn

Reply:
ok